scripts/ErrorReporter/error_report_presenter.py

```diff
@@ -10,20 +10,29 @@
 
 import requests
 
-from ErrorReporter.retrieve_recovery_files import zip_recovery_directory
 from mantid.kernel import ConfigService, ErrorReporter, Logger, UsageService
 
 
 class ErrorReporterPresenter(object):
     SENDING_ERROR_MESSAGE = 'There was an error when sending the report.\nPlease contact mantid-help@mantidproject.org directly'
 
-    def __init__(self, view, exit_code, application='mantidplot'):
+    def __init__(self, view, exit_code, application='mantidplot', traceback=''):
         self.error_log = Logger("error")
         self._view = view
         self._exit_code = exit_code
         self._application = application
+        self._traceback = traceback
         self._view.set_report_callback(self.error_handler)
 
+        if not traceback:
+            traceback_file_path = os.path.join(ConfigService.getAppDataDirectory(), '{}_stacktrace.txt'.format(application))
+            if os.path.isfile(traceback_file_path):
```

Comment:
Similarly we need to make sure we handle errors around the os calls. Can we use `with` for opening the file and I think `open` and `os.rename` need an `OSError` catcher for safety?